services/order.py

```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime = None,
+
+) -> None:
+    order = Order.objects.create(user=get_user_model().
+                                 objects.get(username=username))
+    if date:
+        order.created_at = date
+    [Ticket.objects.create(
+        movie_session=MovieSession.objects.get(id=ticket["movie_session"]),
+        row=ticket["row"],
+        seat=ticket["seat"],
+        order=order
+    )for ticket in tickets]
```

Comment:
you’re using a list comprehension to create Ticket instances. List comprehensions are generally meant for generating lists, not for executing functions with side effects.